src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/JQueryKeyValuePairNormalizer.cs

```diff
@@ -86,7 +86,12 @@ private static string NormalizeJQueryToMvc(StringBuilder builder, string key)
                 else
                 {
                     // Field name. Convert to dot notation.
-                    builder.Append('.');
+                    if (builder.Length != 0)
```

Comment:
Was this a new bug? (assuming it was for when `key == ""`

Reply:
Each `src/` file I changed contained _part_ of the fix needed to get complex and collection `TValue` dictionaries working end-to-end with jQuery-formatted requests.

This part was about cases like `"[][property]"` or `"[property]"` in the request body. Those were previously normalized as `".property"`, not something model binding can use. Now those cases are normalized as `"property"`. An initial non-bracketed key segment and any later key segment are not affected at all. And, `key == string.Empty` can't reach here because `indexOpen == -1` is handled a few lines [above](https://github.com/aspnet/Mvc/pull/7641/files#diff-0513db7554f9855f44c04636b349c404R59).